quickwit/quickwit-ingest/src/ingest_v2/workbench.rs

```diff
@@ -358,11 +536,117 @@ impl IngestSubworkbench {
 
 #[cfg(test)]
 mod tests {
-    use quickwit_proto::ingest::ingester::PersistFailureReason;
-    use quickwit_proto::types::ShardId;
+    use std::time::Duration;
+
+    use quickwit_proto::ingest::ingester::{PersistFailureReason, PersistSubrequest};
+    use quickwit_proto::types::{IndexUid, ShardId, SourceUid};
 
     use super::*;
 
+    #[tokio::test]
```

Comment:
nitpick: I would have tested the `ShardPublishState` object independently.
(Note this is considered bad practise by some company however)